src/nrepl/project-types.ts

```diff
@@ -473,7 +473,10 @@ async function cljCommandLine(connectSequence: ReplConnectSequence, cljsType: Cl
         ...serverPrinterDependencies
     };
     const useMiddleware = [...middleware, ...(cljsType ? cljsMiddleware[cljsType] : [])];
-    const aliasesOption = aliases.length > 0 ? `-M${aliases.join("")}` : '-M';
+    let aliasesOption = aliases.length > 0 ? `-M${aliases.join("")}` : '-M';
+    if (getConfig().jackIn.useDeprecatedAliasFlag) {
+      aliasesOption = aliases.length > 0 ? `-A${aliases.join("")}` : '';
+    }
```

Comment:
I think it would be clearer if we only choose the flag with the option. So something like:
```suggestion
    const aliasesFlag = (getConfig().jackIn.useDeprecatedAliasFlag) ? ['-A', '' : ['-M', '-M'];
    const aliasesOption = aliases.length > 0 ? `${aliasesFlag[0]}${aliases.join("")}` : aliasesFlag[0];
```
(Not tested 😄)